src/transformers/generation_utils.py

```diff
@@ -841,6 +841,80 @@ def compute_transition_beam_scores(
+        if "kwargs" in model_args:
+            model_args |= set(inspect.signature(self.forward).parameters)
+        for key, value in model_kwargs.items():
+            if value is not None and key not in model_args:
+                unused_model_args.append(key)
+
+        if unused_model_args:
+            raise ValueError(
+                f"The following `model_kwargs` are not used by the model: {unused_model_args} (note: typos in the"
+                " generate arguments will also show up in this list)"
+            )
+
+    def _validate_generation_inputs(
+        self,
+        generation_inputs: Dict[str, Any],
+        generation_method_name: str,
+        supporting_objects: List[Callable],
+    ):
+        """
+        Validates the generation inputs for the generation submethod. If there are arguments that were passed but will
+        not be used by the generation method or its supporting objects, an informative exception will be thrown. A
+        supporting object is a class or function that precedes the generation submethod itself.
+        """
+        # Excludes arguments that are handled before calling the submethods
+        unused_keys = ["self"]
+        keys_processed_before_method = [
+            "inputs",
+            "bos_token_id",
+            "max_new_tokens",
+            "decoder_start_token_id",
+            "do_sample",
+            "model_kwargs",
+        ]
+        keys_passed_to_model_kwargs = ["use_cache", "output_attentions", "output_hidden_states"]
+        for key in unused_keys + keys_processed_before_method + keys_passed_to_model_kwargs:
+            generation_inputs.pop(key)
+
+        # Validates that the remainder arguments are used by the generation submethod
+        unused_args = []
+        generation_method = getattr(self, generation_method_name)
+        generation_method_args = set(inspect.signature(generation_method).parameters)
+        for supporting_object in supporting_objects:
+            generation_method_args |= set(inspect.signature(supporting_object).parameters)
+        # Ad hoc replacement of supporting object argument to match the corresponding generation argument
+        if "do_early_stopping" in generation_method_args:
+            generation_method_args.remove("do_early_stopping")
+            generation_method_args.add("early_stopping")
+        for key, value in generation_inputs.items():
+            if value is not None and key not in generation_method_args:
+                unused_args.append(key)
+
+        if unused_args:
+            raise ValueError(
+                f"From the generation arguments, `{generation_method_name}` was triggered. The following arguments are"
+                f" not used by `{generation_method_name}`: {unused_args}. Please remove them from the generation"
+                " arguments or check the documentation for more information."
+            )
```

Comment:
Here is an example of the output for

```python
from transformers import AutoTokenizer, AutoModelForCausalLM

tokenizer = AutoTokenizer.from_pretrained("distilgpt2")
model = AutoModelForCausalLM.from_pretrained("distilgpt2")

prompt = tokenizer(["hello world"], return_tensors="pt")
model.generate(**prompt, num_return_sequences=2, temperature=2.0)
```

<img width="1511" alt="Screenshot 2022-07-21 at 14 32 55" src="https://user-images.githubusercontent.com/12240844/180226232-2dac7b24-2d1d-43fd-8782-5ffdf9d84f34.png">